figures/2.figure2/scripts/figure2.r

```diff
@@ -0,0 +1,373 @@
+    summarise_all(mean)
+# cytokine_values_agg <- as.matrix(cytokine_values_agg)
+
+head(cytokine_values_melted_agg)
+
+# create a matrix of the cytokine values for the clustering and heatmap
+# unmelt the data
+cytokine_values_agg <- reshape2::dcast(cytokine_values_melted_agg, oneb_Metadata_Treatment_Dose_Inhibitor_Dose ~ Cytokine, value.var = "Cytokine_Value")# make oneb_Metadata_Treatment_Dose_Inhibitor_Dose the rownames
+rownames(cytokine_values_agg) <- cytokine_values_agg$oneb_Metadata_Treatment_Dose_Inhibitor_Dose
+# drop the column
+cytokine_values_agg <- subset(cytokine_values_agg, select = -c(oneb_Metadata_Treatment_Dose_Inhibitor_Dose))
+# convert to matrix
+# cytokine_values_agg <- as.matrix(cytokine_values_agg)
+#
+
+cytokine_values_agg
+
+row_dend <- as.dendrogram(hclust(dist(cytokine_values_agg)))
+col_dend <- as.dendrogram(hclust(dist(t(cytokine_values_agg))))
+
+# remove '[NSU]' from the column names
+colnames(cytokine_values_agg) <- gsub("\\[NSU\\]", "", colnames(cytokine_values_agg))
+
+
+cytokine_values_agg <- as.matrix(cytokine_values_agg)
+
+# set plot size
+options(repr.plot.width=25, repr.plot.height=5)
+heatmap_plot_all <- (
+  Heatmap(
+  (cytokine_values_agg),
+  col = brewer.pal(9, "GnBu"),
+  cluster_rows = TRUE,    # Cluster rows
+  cluster_columns = TRUE, # Cluster columns
+  show_row_names = TRUE,  # Show row names
+  show_column_names = TRUE, # Show column names
+  column_names_gp = gpar(fontsize = 10), # Column name label formatting
+  row_names_gp = gpar(fontsize = 12),    # Row name label formatting
+  heatmap_legend_param = list(title = "Legend", at = c(0, 1)),
+  # make the tiles rectangular
+  rect_gp = gpar(col = NA),
+  )
+)
+heatmap_plot_all
+
+# import the tukey test results
+tukey_results_path <- file.path("..","..","2.Nomic_nELISA_Analysis/0.Exploratory_Analysis/PBMC/results/tukey_test_results.csv")
+# read in the data
+tukey_results <- read.csv(tukey_results_path, header = TRUE, sep = ",")
+
+# reload in the cytokine values
+data_path_cytokine_values_melted <- file.path("..","..","2.Nomic_nELISA_Analysis/0.Exploratory_Analysis/PBMC/results/PBMC_all_cytokine_values_per_treatment_per_well_melted.csv")
+cytokine_values_melted <- read.csv(data_path_cytokine_values_melted, header = TRUE, sep = ",")
+# filter out the treatments that are not in the list
+cytokine_values_melted <- cytokine_values_melted[cytokine_values_melted$oneb_Metadata_Treatment_Dose_Inhibitor_Dose %in% list_of_treatments,]
+
+# aggregate the cytokine values across the treatments and cytokine
+cytokine_values_melted_agg <- cytokine_values_melted %>%
+    group_by(oneb_Metadata_Treatment_Dose_Inhibitor_Dose, Cytokine) %>%
+    summarise(Cytokine_Value = mean(Cytokine_Value))
+
+cytokine_values_melted_agg_filtered <- cytokine_values_melted_agg[cytokine_values_melted_agg$Cytokine %in% unique(tukey_results$cytokine),]
+head(cytokine_values_melted_agg_filtered)
+
+# un melt the data cytokine_values_melted_agg_filtered
```

Comment:
```suggestion
# un melt the data cytokine_values_melted_agg_filtered
```
I would recommend giving a bit of a description of what being melted is or what it is to "un melt"

Reply:
Addressed